src/merkle/merkle-tree.ts

```diff
@@ -1,15 +1,17 @@
 import { CompareFunction, MergeFunction, MetadataFunction, Node, PathDirection, TreeMetadata } from './merkle';
 
 /**
- * Merkle tree structure
+ * Merkle tree structure.
+ * Type 'N' is the type of the nodes in the tree. Type 'L' is the type of the leaf nodes specifically,
```

Comment:
Pardon my ignorance here - why was this necessary?

Reply:
As discussed on our call this morning, this is to allow more precise type usage and distinguishing between the type used for the leaf nodes from that used for the intermediate nodes in the tree